events/companies/views.py

```diff
@@ -0,0 +1,234 @@
+        new_company_data = json.loads(request.body.decode())
+        company_form = CompanyForm(new_company_data)
+        errors = company_form.errors
+        user = User.get_by_id(new_company_data.get('admin'))
+        errors = CompanyView.validate_admin_field(user, errors, new_company_data)
+        if not company_form.is_valid() or errors:
+            return JsonResponse({'success': False, 'errors': errors}, status=400)
+        new_company_data['admin'] = user
+        Company.objects.create(**new_company_data)
+        return CREATED
+
+    def put(self, request, company_id):
+        company = Company.get_by_id(company_id)
+        if not company:
+            return COMPANY_NOT_EXISTS
+        if not CompanyView.check_company_rights(request, company):
+                return PERMISSION_DENIED
+        upd_company_data = json.loads(request.body.decode())
+        company_form = CompanyForm(upd_company_data)
+        errors = company_form.errors
+        upd_admin = User.get_by_id(upd_company_data.get('admin'))
+        if upd_admin and upd_admin != company.admin:
+            if not request.user.is_superuser:
+                return PERMISSION_DENIED
+            errors = CompanyView.validate_admin_field(upd_admin, errors, upd_company_data, company)
+        else:
+            upd_admin = company.admin
+        if not company_form.is_valid() or errors:
+            return JsonResponse({'success': False, 'errors': company_form.errors}, status=400)
+        upd_company_data['admin'] = upd_admin
+        Company.objects.filter(pk=company_id).update(**upd_company_data)
+        return CREATED
+
+    def delete(self, request, company_id):
+        company = Company.get_by_id(company_id)
+        if not CompanyView.check_company_rights(request, company):
+                return PERMISSION_DENIED
+        company.delete()
+        return NO_CONTENT
+
+    @staticmethod
+    def check_company_rights(request, company):
+        if not request.user.is_superuser:
+            return company.admin.id == request.user.id
+        return True
+
+    @staticmethod
+    def validate_admin_field(user, errors, data, company=None):
+        if not data.get('admin'):
+            errors['admin'] = ['This field is required']
+        else:
+            if not user:
+                errors['admin'] = ["This user didn't exists"]
+        if not company:
+            if Company.objects.filter(admin=user):
+                errors['admin'] = ['This user is already an admin of another company']
+        else:
+            if Company.objects.filter(admin=user) and company.admin != user:
+                errors['admin'] = ['This user is already an admin of another company']
+        return errors
+
+
+class TeamView(View):
+
+    def get(self, request, company_id, team_id=None):
+        company = Company.get_by_id(company_id)
+        if not company:
+            return COMPANY_NOT_EXISTS
+        if Company.get_user_company(request) != company:
```

Comment:
Looks like there are few more places where you can use `and` instead of two `if`s